sites/friday/src/get-started/here/make-changes.md

```diff
@@ -0,0 +1,108 @@
+   **Secondly**, if you have set up an integration and have enabled `prune_branches` (true by default), then `upsun branch` command will fail at the end of the deployment because the integration will sync with the source and delete the new branch since it doesn’t exist with the canonical source.
+   At the end, you will see:
+   ```bash
+   fatal: couldn't find remote ref feat-a
+   ```
+   {{< /note >}}
+<--->
++++
+title=Using third party provider
++++
+   {{% vendor/name %}} provides a feature called [source integration](integrations/source.html) that allows your {{% vendor/name %}} project to be fully integrated with your external repository.
+   This enables you, as a developer, to use a normal Git workflow (`git add . && git commit -m "message" && git push`) to deploy your environment—with no need to connect to the {{% vendor/name %}} Console.
+
+   Assuming you already activate source integration feature on your project, you need to use the following to create a new branch:
+   ```bash {location="Terminal"}
+   git checkout -b feat-a && git push -u origin feat-a
+   ```
+   This command creates a new local `feat-a` Git branch based on the `main` Git branch
+   and create related environment on {{< vendor/name >}}.
+   The new environment is inactive by default, and when you would activate it, it will inherit the data (service data and assets) of its parent environment (the production environment here).
+
+   {{< note >}}
+   When your local feature is ready, please see how to [activate your environment](/environments/deactivate-environment.md#reactivate-an-environment)
+   {{< /note >}}
+
+{{< /codetabs >}}
+
+
+2. Make changes to your project.
+
+   Depending on the stack you're using, change something within your source code.
+
+3. Commit your changes:
+
+   ```bash {location="Terminal"}
+   git add index.js
+   git commit -m "Update source code"
+   ```
+
+4. Deploy your changes to the `feat-a` environment:
+
+   ```bash {location="Terminal"}
+   {{% vendor/cli %}} push
+   ```
+
+   Note that each environment has its own domain name.
+   To open the url of your new environment, run the following command:
+
+   ```bash {location="Terminal"}
+   {{% vendor/cli %}} environment:url --primary
+   ```
+
+5. Iterate by changing the code, committing, and deploying.
+   When satisfied with your changes, merge them to the main branch,
+   and remove the feature branch:
+
+   ```bash {location="Terminal"}
+   {{% vendor/cli %}} merge
+     Are you sure you want to merge feat-a into its parent, main? [Y/n] y
+   {{% vendor/cli %}} checkout main
+   git pull {{% vendor/cli %}} main
+   {{% vendor/cli %}} environment:delete feat-a
+   git fetch --prune
+   ```
+
+   {{< note >}}
+   Deploying to production was fast because the image built for the `feat-a` environment was reused for your `main` environment.
+   {{< /note >}}
+
+   For a long running branch, to keep the code up-to-date with the main branch, use `git merge main` or `git rebase main`.
```

Comment:
this seems out of place? I assume what we're suggesting is incorporating changes from `main` back into dev branches, but in step 5 we had them switch back to the `main` branch so attempting `merge main` or `rebase main` when you have `main` checked out doesn't do anything.

Reply:
i make it a real 6. step
https://docs.upsun.com.fix-generic-gstarted-7jc6g4y-ucq44jg6ofare.eu-5.platformsh.site/get-started/here/make-changes.html#4-deploy-your-changes
is that ok?

Reply:
see https://github.com/platformsh/platformsh-docs/pull/3554#issuecomment-1949231306